### ISSUE TITLE
Feature request: Option to output linker .map files - ToDo documentation

### ISSUE THREAD

**Issue by AkiMakkonen-AlifSemi** [Open-CMSIS-Pack/devtools#1549]
It would be nice to have a generic way to configure .map output just like the .elf and .bin

https://github.com/Open-CMSIS-Pack/cmsis-toolbox/blob/main/docs/YML-Input-Format.md#output

Currently different toolchains need to have their own linker parameters in cdefault.yaml to accomplish this.

**Comment by ReinhardKeil** [comment]
In the IDE this could be shown like this:

![image](https://github.com/user-attachments/assets/b4f85cc7-f445-4611-b6ae-b3fd2a1eafda)

**Comment by brondani** [comment]
As requested the `map` type was added to the [output](https://github.com/Open-CMSIS-Pack/cmsis-toolbox/blob/main/docs/YML-Input-Format.md#output) node, as well as an equivalent [access sequence](https://github.com/Open-CMSIS-Pack/cmsis-toolbox/blob/main/docs/YML-Input-Format.md#access-sequences) `$map(context)$`. It also appears in the equivalent `output` node in cbuild.yml files.
The filename's format follows the implementation in [cdefault.yml](https://github.com/Open-CMSIS-Pack/devtools/blob/tools/projmgr/2.6.0/tools/projmgr/templates/cdefault.yml), so if such options will still be used (for example via local cdefault.yml) they result in identical command line options and don't break the build.
Note that AC6, GCC, CLANG accept multiple map file options and take the last, while IAR accept multiple options only when they are identical.
On this matter there was a 3yo issue in CMake [#21538](https://gitlab.kitware.com/cmake/cmake/-/issues/21538) that prevents further customization of the map filename due to an orphaned option among the default armlink flags. The [PR](https://gitlab.kitware.com/cmake/cmake/-/merge_requests/10008) to fix that has just been accepted and merged to the 3.32 milestone.

**Comment by jkrech** [comment]
```
  output:
    type:
    - map
```

results in: `*.cbuild.yml`
```
build:
 ...
 output:
    - type: elf
      file: Hello.axf
    - type: map
      file: Hello.axf.map
```
